src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -382,217 +376,212 @@ export class HuiThermostatCard extends LitElement implements LovelaceCard {
     });
   }
 
-  private formatTemp(temps: string[], spaceStepSize: boolean): string {
+  private formatTemp(temps: number[], spaceStepSize: boolean): string {
     temps = temps.filter(Boolean);
+    let stringTemps = temps.map(String);
 
     // If we are sliding the slider, append 0 to the temperatures if we're
     // having a 0.5 step size, so that the text doesn't jump while sliding
     if (spaceStepSize) {
       const stepSize = this._stepSize;
-      temps = temps.map((val) =>
+      stringTemps = stringTemps.map((val) =>
         val.includes(".") || stepSize === 1 ? val : `${val}.0`
```

Comment:
`val.toFixed(1)` ?

Reply:
I suppose stepsize is unlikely to be less than 0.1 (or even 0.5).

Reply:
But again, I think this would be better in `render`.

Reply:
what do you mean that it would be better in render? This is used in multiple places in the file.

Reply:
I mean I think you could have a property for `setTemp` and making the render function do this processing:

Something like (simplified):

```js
${ Array.isarray(this.setTemp)
  ? html`${this.setTemp[0].toFixed(1)} - ${this.setTemp[1].toFixed(1)}`
  : html`${this.setTemp.toFixed(1)}
}
```

Reply:
I get it! :tada: sorry for being dense. Yes, that makes a lot of sense. Thanks.